drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/workarounds/KogitoRuntimeAsyncPackageDataModelOracle.java

```diff
@@ -0,0 +1,182 @@
+    private String[] factTypes;
+    private String[] fqcnNames;
+    private Map<String, String> fqcnNamesMap;
+    private Map<String, Boolean> collectionTypes;
+    private Map<String, ModelField[]> modelFieldsMap;
+
+    @Override
+    public void init(Path resourcePath) {
+        this.resourcePath = resourcePath;
+        this.parametricFieldMap = retrieveParametricFieldMap();
+        this.packageNames = retrievePackageNames();
+        this.factTypes = retrieveFactTypes();
+        this.fqcnNames = retrieveFqcnNames();
+        this.fqcnNamesMap = retrieveFqcnNamesMap();
+        this.collectionTypes = retrieveCollectionTypes();
+        this.modelFieldsMap = retrieveModelFieldsMap();
+    }
+
+    @Override
+    public Path getResourcePath() {
+        return resourcePath;
+    }
+
+    @Override
+    public List<String> getPackageNames() {
+        return packageNames;
+    }
+
+    @Override
+    public String[] getFactTypes() {
+        return factTypes;
+    }
+
+    @Override
+    public String[] getAllFactTypes() {
+        return factTypes;
+    }
+
+    @Override
+    public String[] getInternalFactTypes() {
+        return factTypes;
+    }
+
+    @Override
+    public String[] getExternalFactTypes() {
+        return new String[0];
+    }
+
+    @Override
+    public String getFQCNByFactName(String factName) {
+        return fqcnNamesMap.get(factName);
+    }
+
+    @Override
+    public ModelField[] getFieldCompletions(String factType) {
+        return modelFieldsMap.get(factType);
+    }
+
+    @Override
+    public String getFieldType(String variableClass, String fieldName) {
+        String toReturn = null;
+        ModelField modelField = getModelField(variableClass, fieldName);
+        if (modelField != null) {
+            toReturn = modelField.getType();
+        }
+        return toReturn;
+    }
+
+    @Override
+    public String getFieldClassName(String factName, String fieldName) {
+        String toReturn = null;
+        ModelField modelField = getModelField(factName, fieldName);
+        if (modelField != null) {
+            toReturn = modelField.getClassName();
+        }
+        return toReturn;
+    }
+
+    @Override
+    public String getParametricFieldType(String factType, String fieldName) {
+        String key = factType + "." + fieldName;
+        return parametricFieldMap.get(key);
+    }
+
+    @Override
+    public PackageDataModelOracleBaselinePayload getPackageDataModelOracleBaselinePayload() {
+        PackageDataModelOracleBaselinePayload toReturn = new PackageDataModelOracleBaselinePayload();
+        toReturn.setModelFields(modelFieldsMap);
+        toReturn.setPackageName(packageNames.get(0));
+        toReturn.setCollectionTypes(collectionTypes);
+        toReturn.setFieldParametersType(parametricFieldMap);
+        return toReturn;
+    }
+
+    private Map<String, String> retrieveParametricFieldMap() {
+        // TODO implement retrieval from kogitoResourceContentService
```

Comment:
Can you please create a ticket and refer it here about bean introspection client side? Maybe it already exists

Reply:
@danielezonca I found this: https://issues.redhat.com/browse/KOGITO-665, I think we can refer to this one for SCESIM as well.